src/portable/synopsys/dwc2/dcd_dwc2.c

```diff
@@ -650,6 +658,108 @@ void dcd_edpt_close_all(uint8_t rhport) {
   _allocated_fifo_words_tx = 16;
 }
 
+bool dcd_edpt_iso_alloc(uint8_t rhport, uint8_t ep_addr, uint16_t largest_packet_size)
```

Comment:
we should refactor this to share code with dcd_edpt_open(), since it is quite a duplication. I can do it if you don't mind testing again :)

Reply:
Yes please do it.